### PR TITLE
feat: add har replay support

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes https://github.com/microsoft/playwright-dotnet/issues/2146

### CODE REVIEW

File: src/Playwright/Core/Page.cs [hunk 1]
```diff
@@ -1153,6 +1153,15 @@ private FrameSetInputFilesOptions Map(PageSetInputFilesOptions options)
             };
         }
 
-        public Task RouteFromHARAsync(string har, PageRouteFromHAROptions options = null) => throw new NotImplementedException();
+        public async Task RouteFromHARAsync(string har, PageRouteFromHAROptions options = null)
+        {
+            var harRouter = await HarRouter.CreateAsync(Channel.Connection.LocalUtils, har, options?.NotFound, new()
```
Comment (rwoll) [inline src/Playwright/Core/Page.cs:1158]: Default notFound to abort: https://github.com/microsoft/playwright/blob/ae6f48c4b8bb55db693dfdfd27592d9206e0d46d/packages/playwright-core/src/client/browserContext.ts#L274

File: src/Playwright/Core/HarRouter.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                }).ConfigureAwait(false);
+                return;
+            }
+
+            if (_notFoundAction == HarNotFound.Abort)
+            {
+                await route.AbortAsync().ConfigureAwait(false);
+                return;
+            }
+
```
Comment (rwoll) [inline src/Playwright/Core/HarRouter.cs:93]: Put the no-op error placeholder just to keep things parallel to the other ports.

File: src/Playwright/Core/Page.cs (hunk 1 above)
Comment (mxschmitt) [reply src/Playwright/Core/Page.cs:1158]: I do it inside the HARRouter, but let me move it up a level.

File: src/Playwright/Core/BrowserContext.cs [hunk 3]
```diff
@@ -496,6 +496,15 @@ private Task ExposeBindingAsync(string name, Delegate callback, bool handle = fa
             return Channel.ExposeBindingAsync(name, handle);
         }
 
-        public Task RouteFromHARAsync(string har, BrowserContextRouteFromHAROptions options = null) => throw new NotImplementedException();
+        public async Task RouteFromHARAsync(string har, BrowserContextRouteFromHAROptions options = null)
+        {
+            var harRouter = await HarRouter.CreateAsync(Channel.Connection.LocalUtils, har, options?.NotFound, new()
```
[the remark this answers is not in the document]
Comment (mxschmitt) [reply src/Playwright/Core/BrowserContext.cs:501]: I do it inside the HARRouter, but let me move it up a level.

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**